app/preprints/-components/submit/preprint-state-machine/component.ts

```diff
@@ -221,11 +235,62 @@ export default class PreprintStateMachine extends Component<StateMachineArgs>{
     public async onSubmit(): Promise<void> {
         this.args.resetPageDirty();
 
+        if (this.isNewVersionFlow) {
+            try {
+                const url = this.preprint.links.preprint_versions as string;
```

Comment:
I've changed this workflow so that the new version is created _before_ going in to this workflow, as you had previously said. I implemented this workflow incorrectly, so sorry for the confusion!